api/kubelb.k8c.io/v1alpha1/common_types.go

```diff
@@ -86,14 +86,37 @@ type EndpointAddress struct {
 	Hostname string `json:"hostname,omitempty" protobuf:"bytes,3,opt,name=hostname"`
 }
 
+type Annotations map[string]string
+
+// +kubebuilder:validation:Enum=all;service;ingress;gateway;httproute;grpcroute;tcproute;udproute;tlsroute
+type AnnotatedResource string
+
+const (
+	AnnotatedResourceAll       AnnotatedResource = "all"
```

Comment:
That looks good, do we also want to propagate annotations to envoy pods?

Reply:
I think just for the load balancing resources should be fine. For now, I don't see a good use case where user would need annotations on Envoy Pods.

Reply:
But if something comes up, I think it should be handled via this config instead https://github.com/kubermatic/kubelb/blob/main/api/kubelb.k8c.io/v1alpha1/config_types.go#L45-L85